### DISCUSSION TITLE
Confusing NGA50 value reported by metaquast

### DISCUSSION THREAD

**Discussion by HaploKit** [ablab/quast#174] [Q&A] [answered]
Hi, I am using metaquast for metagenome assembly evaluation, but find a confusing NGA50/NG50 value. The confusing point is that Genome fraction (%)=39.9, which is less than 50%, but NGA50=5456, to my understanding, NGA50 or NG50 should be 'none' according to the definition. The reference genome consists of 5 strains(genomes), each of them is about 10kbp. Did I misunderstand something? Any help would greatly be appreciated.

The assemblies are generated by Canu, there are 3 contigs, the contig length is 9665, 9668, 5457bp.

Here is the command and version I am using：
`quast-5.1.0rc1/metaquast.py -r $ref  --min-contig 500 -o out --unique-mapping -t 16 $fa`

Here is the report:
```
$ cat out/runs_per_reference/ref/report.tsv
Assembly	out.contigs
# contigs (>= 1000 bp)	3
# contigs (>= 5000 bp)	3
# contigs (>= 10000 bp)	0
# contigs (>= 25000 bp)	0
# contigs (>= 50000 bp)	0
Total length (>= 1000 bp)	24790
Total length (>= 5000 bp)	24790
Total length (>= 10000 bp)	0
Total length (>= 25000 bp)	0
Total length (>= 50000 bp)	0
# contigs	3
Largest contig	9668
Total length	24790
Reference length	48492
GC (%)	41.60
Reference GC (%)	41.81
N50	9665
NG50	5457
N90	5457
NG90	-
L50	2
LG50	3
L90	3
LG90	-
# misassemblies	0
# misassembled contigs	0
Misassembled contigs length	0
# local misassemblies	0
# scaffold gap ext. mis.	0
# scaffold gap loc. mis.	0
# unaligned mis. contigs	0
# unaligned contigs	0 + 0 part
Unaligned length	0
Genome fraction (%)	39.901
Duplication ratio	1.281
# N's per 100 kbp	0.00
# mismatches per 100 kbp	4.03
# indels per 100 kbp	100.86
Largest alignment	9667
Total aligned length	24788
NA50	9665
NGA50	5456
NA90	5456
NGA90	-
LA50	2
LGA50	3
LA90	3
LGA90	-
```

**Comment by alexeigurevich** [comment, marked as answer]
Hi, this is a great question!
The explanation is the following. When we are computing NGA50, we just look at **lengths of aligned** fragments and select the one such that all alignments fragments of the same or larger length together sum up to at least 50% of the reference length, which is `48492*0.5 = 24246` bp in your case. Note that the `Total aligned length` is `24788`, which is just above `24246`, so if we take all alignments (i.e., set NGA50 to the shortest alignment length = `5456` bp), then we can exceed 50% of the reference length, and thus NGA50 is not none.

At the same time, when we compute `Genome fraction`, we check the actual alignment coordinates. So, if two alignments overlap, their genome coverage is smaller than just the sum of two alignment lengths. Note `Duplication ratio = 1.281` in your report, which roughly means 28% of overlaps between alignments. A duplication ratio above 1.0 is rather common of metagenomes since many fragments of closely related strains are similar to each other, so that assembly fragments may map to the same region equally well. As a result, due to overlaps, the **unique portion** of the reference genome covered by all alignments is just `39.901 %`, not `24788/48492 ~ 51.1%` as you may expect from `Total aligned length = 24788`.

Side note: I will transfer this issue to the recently opened QUAST Discussion forum!
